### PR TITLE
feat(utils): add indicator & install pyimgur for upload img

### PR DESCRIPTION
**Pull request by zous910814**

### CODE REVIEW

**Comment by zous910814** [comment]
update indicator method to use yfinance api

**Review by a5512167086 — COMMENTED** [review]
Please add commet to explain .gitkeep file. 
ex. Add .gitkeep to keep cache folder for saving img to locale and upload to imgur API
Also use # for comment don't use ``` ```

File: src/utils/Indicator.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        ) / (
+            data['Close'].rolling(window=9).max() -
+            data['Close'].rolling(window=9).min()
+        ) * 100
+        rsv = np.nan_to_num(rsv)
+        self.__data['RSV'] = rsv
+
+    def kv(self):
+        '''
+        當日K值=前一日K值 * 2/3 + 當日RSV * 1/3
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Please use English for comment message

File: src/utils/Indicator.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        kv = [20 for _ in range(8)]
+        ktemp = kv[0]
+        for i in range(len(rsv) - 8):
+            ktemp = ktemp * (2 / 3) + rsv[i + 8] * (1 / 3)
+            kv.append(round(ktemp, 2))
+        self.__data['K'] = kv
+
+    def dv(self):
+        '''
+        當日D值=前一日D值 * 2/3 + 當日K值 * 1/3
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Please use English for comment message

File: src/utils/Indicator.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        dv = [50 for _ in range(8)]
+        dtemp = dv[0]
+        for i in range(len(kv) - 8):
+            dtemp = dtemp * (2 / 3) + kv[i + 8] * (1 / 3)
+            dv.append(round(dtemp, 2))
+        self.__data['D'] = dv
+
+    def macd(self):
+        '''
+        EMA(指數移動平均)：計算MACD時，會先計算長、短天期的指數移動平均線(EMA)
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Please use English for comment message

File: src/utils/Indicator.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        ，一般來說短期常使用12日(n=12)、長期為26日(m=26)
+        DIF = 12日EMA – 26日EMA
+        MACD =快線取9日EMA
+        柱狀圖(直方圖) = 快線–慢線，EMA快慢線相減後，得出來的差額就是在MACD圖形看到的柱狀圖。
+        :return:
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Please simplify those comments and use English too

File: src/utils/Indicator.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    input a company's dataframe to create indicators series
+    :return series
+    '''
+
+    def __init__(self, data):
+        self.__data = data
+
+    def rsv(self):
+        '''
+        rsv = (今日收盤價 - 最近九天的最低價)/(最近九天的最高價 - 最近九天最低價)
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Please use English for comment message

File: src/utils/Indicator.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+        data.index = pd.DatetimeIndex(data['Date'])
+        data = data[data.index > date]
+        mc = mpf.make_marketcolors(up='r', down='g', inherit=True)
+        s = mpf.make_mpf_style(base_mpf_style='yahoo', marketcolors=mc)
+
+        if savefig == True:
+            # 5b,20o,60g,120r,240p
+            mpf.plot(data, style=s, type='candle', volume=True, mav=(5, 20, 60, 120, 240),
+                     savefig='picture/' + str(round(data['證券代號'].values[0])) + '.png')
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Add TODO for markdown we need to add UID library for making file name random
[Doc](https://blog.csdn.net/lanchunhui/article/details/61917841)

File: src/utils/Indicator.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+                plt.bar(row['Date'], row['MACD_histogram'],
+                        width=0.5, color='red')
+            else:
+                plt.bar(row['Date'], row['MACD_histogram'],
+                        width=0.5, color='green')
+
+        plt.legend()
+        plt.title('MACD')
+
+        if savefig == True:
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Don't push debugging code on your PR, remove these condition

File: src/utils/Indicator.py [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+        mc = mpf.make_marketcolors(up='r', down='g', inherit=True)
+        s = mpf.make_mpf_style(base_mpf_style='yahoo', marketcolors=mc)
+
+        if savefig == True:
+            # 5b,20o,60g,120r,240p
+            mpf.plot(data, style=s, type='candle', volume=True, mav=(5, 20, 60, 120, 240),
+                     savefig='picture/' + str(round(data['證券代號'].values[0])) + '.png')
+        else:
+            mpf.plot(data, style=s, type='candle',
+                     volume=True, mav=(5, 20, 60, 120, 240))
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Don't push debugging code on your PR, remove these condition

File: src/utils/Indicator.py [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+        data = data[data.index > date]
+        data['Bias'].plot(color='red')
+        plt.legend()
+        plt.title('Bias')
+
+        if savefig == True:
+            picture = "src/cache/Bias.png"
+            plt.savefig(picture)
+        else:
+            plt.show()
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Don't push debugging code on your PR, remove these condition

File: src/utils/Indicator.py [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+        data['Bollinger_down'].plot(color='green')
+        data['Close'].plot(color='orange')
+        plt.legend()
+        plt.title('Bollinger_band')
+
+        if savefig == True:
+            picture = "src/cache/Bollinger_band.png"
+            plt.savefig(picture)
+        else:
+            plt.show()
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Don't push debugging code on your PR, remove these condition

File: src/utils/Indicator.py [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+        if savefig == True:
+            picture = "src/cache/KD.png"
+            plt.savefig(picture)
+            uploaded_image = im.upload_image(
+                picture, title="Uploaded with PyImgur")
+
+            return uploaded_image.link
+
+        else:
+            plt.show()
```
Comment (a5512167086) [inline src/utils/Indicator.py]: Don't push debugging code on your PR, remove these condition

**Review by a5512167086 — CHANGES_REQUESTED** [review]

**Review by a5512167086 — APPROVED** [review]
Approve

**Pull request merged**